tests/element_test_vectors.rs

```diff
@@ -37,69 +37,56 @@ type SkipList = &'static [&'static str];
 // Once all of the tests are passing, we should work to unify their APIs.
 trait RoundTrip {
     /// Encodes `elements` to a buffer in the specified Ion `format` and then reads them back into
-    /// a `Vec<OwnedElement>` using the provided reader.
+    /// a `Vec<Element>` using the provided reader.
     fn roundtrip<R: ElementReader>(
         elements: &[Element],
         format: Format,
         reader: R,
     ) -> IonResult<Vec<Element>>;
 }
 
-/// These unit structs implement the [RoundTrip] trait and can be passed throughout the
-/// integration tests to dictate which readers/writers should be used in each test.
-struct NativeElementWriterApi;
-
-impl RoundTrip for NativeElementWriterApi {
-    fn roundtrip<R: ElementReader>(
-        elements: &[Element],
-        format: Format,
-        reader: R,
-    ) -> IonResult<Vec<Element>> {
-        // Unlike the C writer, the Rust writer can write into a growing Vec.
-        // No need for an aggressive preallocation.
-        let mut buffer = Vec::with_capacity(2048);
-        match format {
-            Format::Text(kind) => {
-                let writer = match kind {
-                    TextKind::Compact => TextWriterBuilder::new().build(&mut buffer),
-                    TextKind::Lines => TextWriterBuilder::lines().build(&mut buffer),
-                    TextKind::Pretty => TextWriterBuilder::pretty().build(&mut buffer),
-                }?;
-                let mut writer = NativeElementWriter::new(writer);
-                writer.write_all(elements)?;
-                let _ = writer.finish()?;
-            }
-            Format::Binary => {
-                let binary_writer = BinaryWriterBuilder::new().build(&mut buffer)?;
-                let mut writer = NativeElementWriter::new(binary_writer);
-                writer.write_all(elements)?;
-                let _ = writer.finish()?;
-            }
-        };
-
-        reader.read_all(buffer.as_slice())
-    }
```

Comment:
🗺️ I was able to remove the `RoundTrip` trait and its implementations. The one-size-fits-all `serialize` method below performs much of the same logic without needing the trait.

Reply:
🥳